src/realm-settings/key-providers/aes-generated/AESGeneratedForm.tsx

```diff
@@ -0,0 +1,358 @@
+import { useParams, useRouteMatch } from "react-router-dom";
+import { FormAccess } from "../../../components/form-access/FormAccess";
+import { ViewHeader } from "../../../components/view-header/ViewHeader";
+import { convertToFormValues } from "../../../util";
+import { useAlerts } from "../../../components/alert/Alerts";
+import { useRealm } from "../../../context/realm-context/RealmContext";
+
+type AESGeneratedFormProps = {
+  handleModalToggle?: () => void;
+  refresh?: () => void;
+  save?: (component: ComponentRepresentation) => void;
+  editMode?: boolean;
+  providerType: string;
+};
+
+export const AESGeneratedForm = ({
+  editMode,
+  providerType,
+  handleModalToggle,
+  refresh,
+}: AESGeneratedFormProps) => {
+  const { t } = useTranslation("groups");
+  const serverInfo = useServerInfo();
+  const [isKeySizeDropdownOpen, setIsKeySizeDropdownOpen] = useState(false);
+
+  const adminClient = useAdminClient();
+  const { addAlert } = useAlerts();
+  const realm = useRealm();
+
+  const { id } = useParams<{ id: string }>();
+  const { url } = useRouteMatch();
+
+  const providerId = url.split("/").slice(-2)[0];
+
+  const save = async (component: ComponentRepresentation) => {
+    try {
+      if (id) {
+        await adminClient.components.update(
+          { id },
+          {
+            ...component,
+            parentId: realm.realm,
+            providerId: providerId,
+            providerType: "org.keycloak.keys.KeyProvider",
+          }
+        );
+        addAlert(t("realm-settings:saveProviderSuccess"), AlertVariant.success);
+      } else {
+        await adminClient.components.create({
+          ...component,
+          parentId: realm.realm,
+          providerId: providerType,
+          providerType: "org.keycloak.keys.KeyProvider",
+        });
+        handleModalToggle!();
+        addAlert(t("realm-settings:saveProviderSuccess"), AlertVariant.success);
+        refresh!();
+      }
+    } catch (error) {
+      addAlert(
+        t("realm-settings:saveProviderError") +
+          error.response?.data?.errorMessage || error,
+        AlertVariant.danger
+      );
+    }
+  };
+
+  const form = useForm<ComponentRepresentation>({ mode: "onChange" });
+
+  const setupForm = (component: ComponentRepresentation) => {
+    form.reset();
+    Object.entries(component).forEach((entry) => {
+      if (entry[0] === "config") {
+        form.setValue("config.secretSize", entry[1].secretSize[0]);
+
+        form.setValue("config.active", entry[1].active[0]);
+
+        convertToFormValues(entry[1], "config", form.setValue);
+      }
+      form.setValue(entry[0], entry[1]);
+    });
+  };
+
+  useFetch(
+    async () => {
+      if (editMode) return await adminClient.components.findOne({ id: id });
+    },
+    (result) => {
+      if (result) {
+        setupForm(result);
+      }
+    },
+    []
+  );
+
+  const allComponentTypes = serverInfo.componentTypes![
+    "org.keycloak.keys.KeyProvider"
+  ];
```

Comment:
`componentTypes` can be `undefined`, which might cause runtime errors. Let's use optional chaining instead.

```suggestion
  const allComponentTypes = serverInfo.componentTypes?.[
    "org.keycloak.keys.KeyProvider"
  ];
```

If we want to ensure the value is always present we can default to an empty array.

```ts
const allComponentTypes = serverInfo.componentTypes?.["org.keycloak.keys.KeyProvider"] ?? [];
```